src/ao_event.lua

```diff
@@ -0,0 +1,115 @@
+-- Factory function for creating an "AOEvent"
+local function AOEvent(initialData)
```

Comment:
i haven't seen lua classes created this way - the most common pattern i've seen is more like

```
-- Define a class
local MyClass = {}
MyClass.__index = MyClass

-- Constructor for the class
function MyClass:new(name, age)
    local instance = setmetatable({}, MyClass) -- Set the metatable for the new object
    instance.name = name
    instance.age = age
    return instance
end

-- A method for the class
function MyClass:greet()
    print("Hello, my name is " .. self.name .. " and I am " .. self.age .. " years old.")
end

-- Create an instance of the class
local person = MyClass:new("John", 30)
person:greet()  -- Output: Hello, my name is John and I am 30 years old.
```

but regardless - we can look to use classes more for certain objects (gateways, records) to put some protections around accessing/modifying properties directly)